### PR TITLE
configMap and secret guideline

### PR DESCRIPTION
**Pull request by mfranczy**
Guideline on how to use configMap and secret in KubeVirt.
Related to https://github.com/kubevirt/kubevirt/pull/1474

### CODE REVIEW

File: workloads/virtual-machines/disks-and-volumes.md [hunk 1]
```diff
@@ -373,3 +375,98 @@ spec:
...
+    - name: configmap-volume
+      configMap:
+        configMapName: app-config
+    - name: cloudinit-volume
+      cloudInitNoCloud:
+        userData: |
+          #!/bin/sh
+
+          # use a disk serial number to mount the ConfigMap
+          "sudo mount /dev/$(lsblk --nodeps -no name,serial | grep CVLY623300HK240D | cut -f1 -d' ') /blub"
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: Did you try the example? I think you need  to at least also have a `mkdir /blub`.

Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: Real examples are added. I tested them.

**Comment by mfranczy** [comment]
Real examples are added with a note about update propagation.

File: workloads/virtual-machines/disks-and-volumes.md [hunk 2]
```diff
@@ -373,3 +375,120 @@ spec:
...
+        #cloud-config
+        password: fedora
+        chpasswd: { expire: False }
+        bootcmd:
+          # mount the ConfigMap
+          - "mkdir /mnt/app-config"
+          - "mount /dev/$(lsblk --nodeps -no name,serial | grep CVLY623300HK240D | cut -f1 -d' ') /mnt/app-config"
+    name: cloudinitvolume
+  - configMap:
+      configMapName: app-config
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: I think that the examples need to be adjusted to align with the latest code changes. @mfranczy can that be?

Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: Done. I added a note about ```items``` as well.

File: workloads/virtual-machines/disks-and-volumes.md [hunk 3]
```diff
@@ -373,3 +375,124 @@ spec:
 
 Ephemeral VMs very often come with read-only root images and limited tmpfs space. In many cases this is not enough to install application dependencies and provide enough disk space for the application data. While this data is not critical and thus can be lost, it is still needed for the application to function properly during its lifetime. This is where an `emptyDisk` can be useful. An emptyDisk is often used and mounted somewhere in `/var/lib` or `/var/run`.
 
+### configMap
+
+A `configMap` is a reference to a [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `configMap` users can use a `cloudInit` and disk serial number. The `configMapName` needs to be specified for a reference to the created kubernetes `ConfigMap` with the same name.
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: `configMapName` is now just `name` I think. Maybe make the last sentence a little bit simpler.

File: workloads/virtual-machines/disks-and-volumes.md [hunk 4]
```diff
@@ -373,3 +375,124 @@ spec:
...
+
+> **Note:** Currently, the ConfigMap update propagation is not supported. After the update, only a pod will be aware of changes, not running VMI.
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: not running VMIs

File: workloads/virtual-machines/disks-and-volumes.md [hunk 5]
```diff
@@ -373,3 +375,124 @@ spec:
...
+    name: cloudinitvolume
+  - configMap:
+      name: app-config
+    name: configmap-volume
+status: {}
+```
+
+### secret
+
+A `secret` is a reference to a [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `secret` users can use a `cloudInit` and disk serial number. The `secretName` needs to be specified for a reference to the created kubernetes `Secret` with the same name.
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: can use `cloudInit` and the disks serial number.

File: workloads/virtual-machines/disks-and-volumes.md [hunk 6]
```diff
@@ -373,3 +375,124 @@ spec:
...
 
+### configMap
+
+A `configMap` is a reference to a [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `configMap` users can use a `cloudInit` and disk serial number. The `configMapName` needs to be specified for a reference to the created kubernetes `ConfigMap` with the same name.
+
+> **Note:** Currently, the ConfigMap update propagation is not supported. After the update, only a pod will be aware of changes, not running VMI.
+
+> **Note:** Due to the Kubernetes CRD [issue](https://github.com/kubernetes/kubernetes/issues/68466), you cannot control the paths within the volume where ConfigMap keys are projected.
```
Comment (rmohr) [inline workloads/virtual-machines/disks-and-volumes.md]: Due to a Kubernetes CRD issue

**Comment by rmohr** [comment]
@stu-gott could you do a native-speaker check too?

File: workloads/virtual-machines/disks-and-volumes.md (hunk 3 above)
Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: aah yeah, i just forgot about that, sorry, Changing...

Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: Maybe the last sentence could be like this:

The `name` has to be set for a reference to the created Kubernetes `ConfigMap`.

Sounds better and it's clear, what do you think?

Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: @stu-gott ^, is it ok? Any comment?

File: workloads/virtual-machines/disks-and-volumes.md [hunk 7]
```diff
@@ -528,3 +530,124 @@ Change the namespace to suite your installation.
...
+      name: app-config
+    name: configmap-volume
+status: {}
+```
+
+### secret
+
+A `secret` is a reference to a [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `secret` users can use `cloudInit` and the disks serial number. The `secretName` needs to be set for a reference to the created kubernetes `Secret`.
+
+> **Note:** Currently, the Secret update propagation is not supported. After the update, only a pod will be aware of changes, not running VMIs.
```
[the remark this answers is not in the document]
Comment (mfranczy) [reply workloads/virtual-machines/disks-and-volumes.md]: @stu-gott fixed, thanks for a review.
@rmohr do you think I should add something or change?

**Comment by fabiand** [comment]
Thank you!

**Pull request merged**